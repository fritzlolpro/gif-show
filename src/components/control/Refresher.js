import React from 'react'

const Refresher = ({onRefresh}) => {

    return (
        <button className='refresher=button' onClick={onRefresh}><i className="material-icons">sync</i></button>
    )
}

export default Refresher