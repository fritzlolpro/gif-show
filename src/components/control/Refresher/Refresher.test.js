import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Refresher from './Refresher'

Enzyme.configure({adapter: new Adapter()})

describe('Refresher', () => {

    it('Defined', () => {
        expect(Refresher).toBeDefined();
    });

    it('Renders', () => {
        const tree = shallow(<Refresher/>);
        expect(tree).toMatchSnapshot();
    });

    it('Clicks', () => {
        const mockRefresh = jest.fn()
        const refresher = shallow(<Refresher onRefresh={mockRefresh}/>)

        refresher.simulate('click')

        expect(mockRefresh).toHaveBeenCalled()
        expect(mockRefresh.mock.calls.length).toEqual(1)
    });

})
