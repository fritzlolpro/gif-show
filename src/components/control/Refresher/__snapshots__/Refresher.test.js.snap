// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Refresher Renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Refresher />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <i
        className="material-icons"
>
        sync
</i>,
      "className": "refresher=button",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "sync",
        "className": "material-icons",
      },
      "ref": null,
      "rendered": "sync",
      "type": "i",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="material-icons"
>
          sync
</i>,
        "className": "refresher=button",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "sync",
          "className": "material-icons",
        },
        "ref": null,
        "rendered": "sync",
        "type": "i",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
